client/app/collections/filtered.coffee

```diff
@@ -4,12 +4,22 @@ ContactViewModel = require 'views/models/contact'
 
 tagRegExp = search.pattern 'tag'
 app       = null
-scores    = new WeakMap()
+scores    = new Map()
+
+
+Math.median = (array) ->
+    sorted = array.sort (a, b) -> a - b
+    len = array.length
+
+    if len % 2
+        sorted[Math.floor len / 2]
+    else
+        (sorted[len / 2] + sorted[len / 2 - 1]) / 2
```

Comment:
You can simplify those 4 lines with:

```coffee
sorted[Math.floor len / 2]
```

It gives a median for both cases (the array has an even number of items, or an odd number).